pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/table/DynamicPulsarDeserializationSchema.java

```diff
@@ -288,16 +288,16 @@ private void emitRow(@Nullable GenericRowData physicalKeyRow, @Nullable GenericR
                     rowKind,
                     physicalArity + metadataArity);
 
-            for (int keyPos = 0; keyPos < keyProjection.length; keyPos++) {
-                assert physicalKeyRow != null;
-                producedRow.setField(keyProjection[keyPos], physicalKeyRow.getField(keyPos));
-            }
-
             if (physicalValueRow != null) {
                 for (int valuePos = 0; valuePos < valueProjection.length; valuePos++) {
                     producedRow.setField(valueProjection[valuePos], physicalValueRow.getField(valuePos));
                 }
             }
+            
+             for (int keyPos = 0; keyPos < keyProjection.length; keyPos++) {
+                assert physicalKeyRow != null;
```

Comment:
Should we just use `Preconditions.checkNotNull(physicalKeyRow)`
This would throw a `NullPointerException` instead of a puzzled AssertionError.

Reply:
physicalKeyRow will not throw an exception here. i mean physicalValueRow will use a null value to set the key pos field value. so we should use physicalKeyRow.getField(keyPos) to correct it later. that't why i  Change the order of two pieces of code